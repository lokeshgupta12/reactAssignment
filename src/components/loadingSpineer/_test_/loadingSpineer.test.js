import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import LoadingSpinner from '../index';
import Adapter from 'enzyme-adapter-react-16';

// Configure Enzyme adapter for Jest
Enzyme.configure({ adapter: new Adapter() });

describe('<LoadingSpinner /> component', () => {

  it('should match the snapshot', () => {
    const tree = renderer.create(<LoadingSpinner enableLoading={true}/>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the HTML Correctly', () => {
    const wrapper = shallow(<LoadingSpinner enableLoading={true}/>);

    expect(wrapper.find('div').length).toBe(3);
  });
})