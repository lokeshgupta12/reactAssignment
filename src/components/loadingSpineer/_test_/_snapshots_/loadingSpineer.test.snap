// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> component should match the snapshot 1`] = `
  <div className="overlay">
      <div className="overlay__inner">
        <div className="overlay__content"><span className="spinner"></span></div>
      </div>
      </div> 
`;